src/main/java/repositories/TicketDAO.java

```diff
@@ -0,0 +1,88 @@
+package repositories;
+
+import models.Ticket;
+import models.TicketType;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TicketDAO {
+
+    private Connection connection;
+
+    public TicketDAO(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void addTicket(Ticket ticket){
+        String sqlCommand = "INSERT INTO ticket (id, user_id, ticket_type, creation_date) VALUE (?, ?, ?, ?)";
+        try(PreparedStatement statement = connection.prepareStatement(sqlCommand)) {
```

Comment:
The connection is `NULL` here. You need to create it explicitly.

1. add connection configuration in separate `package config ConnectionConfig` class
2.  into this `ConnectionConfig` class create connection in method, for example:
```java
    public static Connection getConnection() throws SQLException {
        return DriverManager.getConnection(jdbcUrl, username, password );
    }
```
3. extract url, username and password into separate text file, put it into resources and read values from there
4. Instead of creating Connection in constructor, open it in in your `TicketDAO` or `UserDAO`  classes with `try-with-resources` construction:
```java
try (Connection connection = ConnectionConfig .getConnection();
      PreparedStatement statement = connection.prepareStatement(sqlCommand)) { ... }
```